app/routes/main/athleteDetails/AthleteDetails.js

```diff
@@ -19,18 +20,17 @@ class AthleteDetails extends Component {
     }).isRequired
   };
 
+  static getPerformanceDetailsLength(props) {
+    return idx(props, _ => _.performance.details.length);
+  }
+
   render() {
     const { athlete } = this.props;
-    // TODO add idx https://github.com/facebookincubator/idx
     return (
       <ScrollView style={[styles.scroll]} showsVerticalScrollIndicator={false}>
         <AthleteCard athlete={athlete} />
         <AthleteDetailsCard
-          rendered={
-            athlete.performance &&
-            athlete.performance.details &&
-            athlete.performance.details.length > 0
-          }
+          rendered={AthleteDetails.getPerformanceDetailsLength(athlete) > 0}
```

Comment:
data should be selected in selector (import selector from "./selector";)

Reply:
we have to talk! not 100% agree with that statement. 
details is a nested property of athlete. 
athlete property is computed and provided by selector. 
View is in charge of dealing with it. root property and nested properties.
(not sure to be crystal clear)

Reply:
you are not wrong but i think that s because we are still wrong 

i now believe that there is an issue arround the name of AthleteDetailsCard and his props
i believe it should be
<AthletePerformancesCard performances={perfomances} .../>

what do you think